pallets/gear/src/lib.rs

```diff
@@ -793,42 +806,48 @@ pub mod pallet {
         /// Returns true if a program has been successfully initialized
         pub fn is_initialized(program_id: ProgramId) -> bool {
             ProgramStorageOf::<T>::get_program(program_id)
-                .map(|(p, _bn)| p.is_initialized())
+                .map(|program| program.program.is_initialized())
                 .unwrap_or(false)
         }
 
         /// Returns true if id is a program and the program has active status.
         pub fn is_active(program_id: ProgramId) -> bool {
             ProgramStorageOf::<T>::get_program(program_id)
-                .map(|(p, _bn)| p.is_active())
+                .map(|program| program.program.is_active())
                 .unwrap_or_default()
         }
 
         /// Returns true if id is a program and the program has terminated status.
         pub fn is_terminated(program_id: ProgramId) -> bool {
             ProgramStorageOf::<T>::get_program(program_id)
-                .map(|(p, _bn)| p.is_terminated())
+                .map(|program| program.program.is_terminated())
                 .unwrap_or_default()
         }
 
         /// Returns true if id is a program and the program has exited status.
         pub fn is_exited(program_id: ProgramId) -> bool {
             ProgramStorageOf::<T>::get_program(program_id)
-                .map(|(p, _bn)| p.is_exited())
+                .map(|program| program.program.is_exited())
                 .unwrap_or_default()
         }
 
+        /// Returns true if there is a program with the specified id (it may be paused).
+        pub fn program_exists(program_id: ProgramId) -> bool {
+            ProgramStorageOf::<T>::program_exists(program_id)
+                || PausedProgramStorageOf::<T>::paused_program_exists(&program_id)
+        }
+
         pub fn get_block_number(program_id: ProgramId) -> <T as frame_system::Config>::BlockNumber {
             ProgramStorageOf::<T>::get_program(program_id)
-                .map(|(_p, bn)| bn)
+                .map(|program| program.block_number)
```

Comment:
```suggestion
                .map(|program| program.expiration)
```

remove block number from exited and terminated program